yarn-project/aztec-nr/authwit/src/entrypoint/app.nr

```diff
@@ -0,0 +1,77 @@
+use dep::aztec::context::PrivateContext;
+use dep::aztec::protocol_types::{
+  constants::GENERATOR_INDEX__SIGNATURE_PAYLOAD,
+  hash::pedersen_hash,
+  traits::{Hash, Serialize}
+};
+
+use crate::entrypoint::function_call::{FunctionCall, FUNCTION_CALL_SIZE_IN_BYTES};
+
+// FUNCTION_CALL_SIZE * ACCOUNT_MAX_CALLS + 1
+global APP_PAYLOAD_SIZE: Field = 17;
+// FUNCTION_CALL_SIZE_IN_BYTES * ACCOUNT_MAX_CALLS + 32
+global APP_PAYLOAD_SIZE_IN_BYTES: Field = 420;
+
+global ACCOUNT_MAX_CALLS: Field = 4;
+
+// Note: If you change the following struct you have to update default_entrypoint.ts
+// docs:start:entrypoint-struct
+struct AppPayload {
+  function_calls: [FunctionCall; ACCOUNT_MAX_CALLS],
+  nonce: Field,
+}
+// docs:end:entrypoint-struct
+
+impl Serialize<APP_PAYLOAD_SIZE> for AppPayload {
+  // Serializes the entrypoint struct
+  fn serialize(self) -> [Field; APP_PAYLOAD_SIZE] {
+    let mut fields: BoundedVec<Field, APP_PAYLOAD_SIZE> = BoundedVec::new(0);
+    for call in self.function_calls {
+      fields.extend_from_array(call.serialize());
+    }
+    fields.push(self.nonce);
+    fields.storage
+  }
+}
+
+impl Hash for AppPayload {
+  fn hash(self) -> Field {
+    pedersen_hash(
+      self.serialize(),
+      GENERATOR_INDEX__SIGNATURE_PAYLOAD
+    )
+  }
+}
+
+impl AppPayload {
+  // Serializes the payload as an array of bytes. Useful for hashing with sha256.
+  fn to_be_bytes(self) -> [u8; APP_PAYLOAD_SIZE_IN_BYTES] {
+    let mut bytes: BoundedVec<u8, APP_PAYLOAD_SIZE_IN_BYTES> = BoundedVec::new(0);
+
+    for i in 0..ACCOUNT_MAX_CALLS {
+      bytes.extend_from_array(self.function_calls[i].to_be_bytes());
+    }
+    bytes.extend_from_array(self.nonce.to_be_bytes(32));
+
+    bytes.storage
+  }
+
+  // Executes all private and public calls
+  // docs:start:entrypoint-execute-calls
```

Comment:
these docs macros seem to be copied from somewhere - let's make sure we also update the docs to point to the right macro
(no need to talk about `app_payload, fee_payload` in docs yet but just fix the macros to prevent it from breaking)

Reply:
I have updated the docs markers and the docs mentioning `EntrypointPayload`. @rahul-kothari would you have another look? :D